app/database/itemStore.go

```diff
@@ -64,7 +74,19 @@ func (s *ItemStore) HasManagerAccess(user AuthUser, itemID int64) (found bool, a
 
 // IsValidHierarchy gets an ordered set of item ids and returns whether they forms a valid item hierarchy path from a root
 func (s *ItemStore) IsValidHierarchy(ids []int64) (bool, error) {
-	return false, nil
+	if len(ids) == 0 {
+		return false, nil
+	}
+
+	if valid, err := s.checkIfItemIsRoot(ids[0]); !valid || err != nil {
+		return false, err
```

Comment:
shouldn't you return `valid, err` instead? here you are turning something valid but with err (don't know for what reason) into something not valid

Reply:
> shouldn't you return `valid, err` instead? here you are turning something valid but with err (don't know for what reason) into something not valid

Actually, we never return true together with error, but I'll fix this anyway.